src/state/reducers/favoritesReducer.js

```diff
@@ -0,0 +1,31 @@
+import * as actionTypes from '../actions/actionTypes';
+
+const initialState = [];
+
+export const favoritesReducer = (state = initialState, action) => {
+  switch(action.type) {
+    case actionTypes.ADD_TO_FAVORITE:
+      const recipe = {
+        ...action.item,
+        addedToFavorite: true
+      }
+      state.map(item => {
```

Comment:
you should use `some` function here

Reply:
you will check whether recipe is already in your favorites, if yes - return state, and update state otherwise